spec/factories/comments.rb

```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :comment, class: Comment do
+    association :commentable, factory: :company
+    association :user, factory: :user
+    tenant_id { create(:user).tenant_id }
```

Comment:
this is actually wrong here

you have association to user
and comment.tenant_id will be different from comment.user.tenant_id

FIX IT
use callbacks in factory_bot

Reply:
add spec

expect(comment.tenant). to eq(comment.user.tenant)